nntrainer/layers/conv2d_layer.cpp

```diff
@@ -438,16 +439,57 @@ void Conv2DLayer::calcDerivative(RunLayerContext &context) {
   /// for each batch
   /// filter_kernel^T X derivaitive  -> column matrix
   /// col2im(column matrix) to reconstruct the original image
-  Tensor &col2im_result = context.getTensor(wt_idx[ConvParams::inter_result]);
-  col2im_result.reshape(calcCol2ImOutputDim(derivative.getDim(), filter_dim));
 
-  for (unsigned int b = 0; b < derivative.batch(); ++b) {
-    Tensor deriv_sub = derivative.getBatchSlice(b, 1);
-    Tensor in_deriv_sub = input_derivative.getBatchSlice(b, 1);
-    deriv_sub.reshape({filter_size, derivative.width() * derivative.height()});
+  unsigned int num_threads = NNTR_NUM_THREADS;
+
+  if (num_threads > derivative.batch())
+    num_threads = 1;
+
+  if (num_threads > 1) {
+    auto dowork = [&](size_t s, size_t e, void *user_data) {
```

Comment:
If we remove 'void *user_data' in lambda, 'nullptr' at line 473 can be removed.
Currently user_data is not being used in the lambda body. Did you put it for future use?

Reply:
yes. for later.

Reply:
thank you.